src/map-caption.js

```diff
@@ -0,0 +1,43 @@
+import './leaflet.js'; 
+import './mapml.js'; 
+
+/* 
+implemented for both mapml-viewer and web-map; however web-map does not focus on map element in the browser resulting in NVDA 
+not being able to read out map-caption and stating that it's an interactive map region
+*/
+export class MapCaption extends HTMLElement {
+    constructor() {
+        super();
+        // calls MutationObserver; needed to observe changes to content between <map-caption> tags and update to aria-label
+        this.observer = new MutationObserver(() => {
+            if (document.querySelector('map-caption').innerHTML == this.parentElement.getAttribute('aria-label'))
+            {
+                this.parentElement.setAttribute('aria-label', this.textContent); 
+            }
+        });
+    }
+    // function to retrieve caption content 
+    get ariaLabel() {
```

Comment:
I am not sure if we need a getter for map-caption as people can just use $map-caption.innerHTML, just like regular HTML. I am sure we will need getters and setters in the future though as map-caption gets more complex!

Reply:
Also noticed that changing the innerHTML for map-caption does not modify the aria-label? I believe this was working in your previous commits? Ex. 
`$map-caption.innerHTML = "test"` changes the map-caption's innerHTML but does not modify the aria-label on mapml-viewer.